org.apache.batik.transcoder.TranscoderException: null

### Type of JetBrains IDE (IntelliJ, PyCharm, etc.) and version
IntelliJ IDEA 2023.1 (build IU-231.8109.175)

### Operating System 
Linux 6.2.9-200.fc37.x86_64 (amd64)

### TeXiFy IDEA version
0.7.28

### Description

This happens while trying to preview any equation and does so regardless of document size.
As a test I used
```latex
\documentclass[13pt]{article}

\begin{document}
  
  $\forall n, n'\colon n' = succ(n) \Rightarrow h(n) \leq 1 + h(n')$
  
\end{document}
```
as a minimal reproducer and then simply pressed CTRL + SHIFT + X while the carret was within the equation.
This is probably an issue with my environment/how IntelliJ and/or how the plugin is installed so I hope it is somehow reproducible for you.

### Stacktrace
```
org.apache.batik.transcoder.TranscoderException: null
Enclosed Exception:
Could not write PNG file because no WriteAdapter is availble
	at org.apache.batik.transcoder.image.ImageTranscoder.transcode(ImageTranscoder.java:128)
	at org.apache.batik.transcoder.XMLAbstractTranscoder.transcode(XMLAbstractTranscoder.java:143)
	at org.apache.batik.transcoder.SVGAbstractTranscoder.transcode(SVGAbstractTranscoder.java:139)
	at nl.hannahsten.texifyidea.action.preview.JlatexmathPreviewer.saveSvgAsPng(JlatexmathPreviewer.kt:79)
	at nl.hannahsten.texifyidea.action.preview.JlatexmathPreviewer.preview(JlatexmathPreviewer.kt:36)
	at nl.hannahsten.texifyidea.action.preview.PreviewFormUpdater.compilePreview(PreviewFormUpdater.kt:66)
	at nl.hannahsten.texifyidea.action.preview.PreviewAction.displayPreview(PreviewAction.kt:89)
	at nl.hannahsten.texifyidea.action.preview.ShowEquationPreview.actionPerformed(ShowEquationPreview.kt:32)
	at nl.hannahsten.texifyidea.action.EditorAction.actionPerformed(EditorAction.kt:70)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.doPerformActionOrShowPopup(ActionUtil.java:333)
	at com.intellij.openapi.keymap.impl.ActionProcessor.performAction(ActionProcessor.java:47)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher$1.performAction(IdeKeyEventDispatcher.java:585)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.lambda$doPerformActionInner$9(IdeKeyEventDispatcher.java:707)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:94)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.lambda$doPerformActionInner$10(IdeKeyEventDispatcher.java:707)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareWithCallbacks(ActionUtil.java:356)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.doPerformActionInner(IdeKeyEventDispatcher.java:704)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.java:648)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.java:596)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processActionOrWaitSecondStroke(IdeKeyEventDispatcher.java:480)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.inInitState(IdeKeyEventDispatcher.java:469)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.dispatchKeyEvent(IdeKeyEventDispatcher.java:225)
	at com.intellij.ide.IdeEventQueue.dispatchKeyEvent(IdeEventQueue.kt:598)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:568)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:68)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:348)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:787)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:348)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:343)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:994)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:994)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$4(IdeEventQueue.kt:343)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:831)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:385)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: org.apache.batik.transcoder.TranscoderException: Could not write PNG file because no WriteAdapter is availble
	at org.apache.batik.transcoder.image.PNGTranscoder.writeImage(PNGTranscoder.java:124)
	at org.apache.batik.transcoder.image.ImageTranscoder.transcode(ImageTranscoder.java:126)
	... 42 more

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,8 +3,10 @@
 ## [Unreleased]
 
 ### Added
+* Update minimum required IntelliJ version to 2023.1
 
 ### Fixed
+* Fix equation preview TranscoderException
 
 ## [0.7.28] - 2023-04-01
 Welcome to TeXiFy IDEA 0.7.28! This release fixes some exceptions, adds comment folding and improves run configuration performance even more.

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -107,6 +107,11 @@ dependencies {
 
     // LaTeX rendering for preview
     implementation("org.scilab.forge:jlatexmath:1.0.7")
+    // https://stackoverflow.com/questions/11677572/dealing-with-xerces-hell-in-java-maven
+    implementation("org.apache.xmlgraphics:batik-codec:1.16") {
+        exclude("xml-apis", "xml-apis")
+        exclude("xml-apis", "xml-apis-ext")
+    }
 
     // Test dependencies
     // No version specified, it equals the kotlin version

